src/microbe_stage/editor/MicrobeEditorAction.cs

```diff
@@ -19,17 +19,17 @@ public class MicrobeEditorAction : ReversibleAction
     public IMicrobeEditorActionData Data;
 
     [JsonProperty]
-    private readonly Action<MicrobeEditorAction> redo;
+    private readonly Action<MicrobeEditorAction, MicrobeSpecies> redo;
```

Comment:
I don't like the species being passed through here. Instead the callback should use MicrobeEditorAction.Editor after making that public (but still read only).